File: tests/java/src/main/java/com/escape/tests/cases/Case050ScoreLedger.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 050: score ledger payload for downstream business processing. */
public class Case050ScoreLedger {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: score ledger records and prepare transport-ready payload.
        String taskName = "score_ledger";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "ledger");
        payload.put("stage", "score");
        payload.put("input", raw);
        // SAFE: payload remains local; only primitive summary string is returned.
        return payload.get("task") + ":" + payload.get("input");
    }
}

File: tests/go/cases/case_050_score_ledger.go
package escape_tests

var retainedCase050 = []map[string]string{}

func Case050ScoreLedger(input string) string {
    // Task case 050: score ledger payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "score_ledger",
        "entity": "ledger",
        "stage": "score",
        "input": raw,
    }
    // SAFE: payload remains local; only primitive summary string is returned.
    return payload["task"] + ":" + payload["input"]
}
